### PR TITLE
Feast IAP acquisitions: Part 3

### PR DESCRIPTION
**Pull request by shtukas**
Project: Feast subscription data for android and iOS

Part 1: https://github.com/guardian/mobile-purchases/pull/1667
Part 2: https://github.com/guardian/mobile-purchases/pull/1695

Part 3:
In this part we implement the logic of the processing of inserted subscriptions, if they are feast subscriptions.

### CODE REVIEW

**Comment by shtukas** [comment]
Screenshot 2024-11-11 at 18 39 21

**Comment by shtukas** [comment]
Screenshot 2024-11-11 at 18 40 04

**Comment by shtukas** [comment]
The two previous comments help confirming that the logging works as intended, but also provide a chance to see the schema.

File: typescript/src/feast/acquisition-events/router.ts [hunk 1]
```diff
@@ -1,23 +1,143 @@
...
+        
+        // We are only interested in feast subscriptions
+        if (!isFeast) {
+            console.log(`Skipping non Feast subscription ${subscriptionId}`);
+            return;
+        }
+
+        const result = await processAcquisition(subscription);
+        if (!result) {
+            // We are exiting but TODO: we are going to write to the deadl letter queue.
```
Comment (MaelGNM) [inline typescript/src/feast/acquisition-events/router.ts:61]: Tiny typo
```suggestion
            // We are exiting but TODO: we are going to write to the dead letter queue.
```

File: typescript/src/feast/acquisition-events/router.ts [hunk 2]
```diff
@@ -1,23 +1,143 @@
...
+        let emptySubscription = new ReadSubscription();
+        emptySubscription.setSubscriptionId(subscriptionId);
+
+        let subscription: Subscription;
+
+        try {
+            subscription = await dynamoMapper.get(emptySubscription);
+        } catch (error) {
+            console.log(`[d2c0251e] Subscription ${subscriptionId} record not found in the subscriptions table. Error: `, error);
+            // We are exiting but TODO: we are going to write to the deadl letter queue.
```
Comment (MaelGNM) [inline typescript/src/feast/acquisition-events/router.ts:45]:
```suggestion
            // We are exiting but TODO: we are going to write to the dead letter queue.
```

File: typescript/src/feast/acquisition-events/router.ts [hunk 3]
```diff
@@ -1,23 +1,143 @@
...
+        
+        let emptySubscription = new ReadSubscription();
+        emptySubscription.setSubscriptionId(subscriptionId);
+
+        let subscription: Subscription;
+
+        try {
+            subscription = await dynamoMapper.get(emptySubscription);
+        } catch (error) {
+            console.log(`[d2c0251e] Subscription ${subscriptionId} record not found in the subscriptions table. Error: `, error);
```
Comment (MaelGNM) [inline typescript/src/feast/acquisition-events/router.ts:44]: What kind of errors might we see here? I think saying "record not found" might be a bit confusing if the error could be something like a network issue.

**Review by MaelGNM — APPROVED** [review]
I left a couple of nitpicky (not blocking) comments about comments/logs but the logic looks good.

File: typescript/src/feast/acquisition-events/router.ts (hunk 3 above)
Comment (shtukas) [reply typescript/src/feast/acquisition-events/router.ts:44]: That's a good point. Bad logging. Let me update the string to only display the error :)

**Pull request merged**